ui/shared/TransactionPreview.qml

```diff
@@ -53,15 +55,39 @@ Item {
                     id: imgFromWallet
                     sourceSize.height: 18
                     sourceSize.width: 18
-                    anchors.right: parent.right
+                    anchors.right: fromArrow.visible ? fromArrow.left : parent.right
+                    anchors.rightMargin: fromArrow.visible ? Style.current.padding : 0
                     anchors.verticalCenter: parent.verticalCenter
                     fillMode: Image.PreserveAspectFit
                     source: "../app/img/walletIcon.svg"
+                    ColorOverlay {
+                        anchors.fill: parent
+                        source: parent
+                        color: root.fromAccount ? root.fromAccount.iconColor : Style.current.blue
+                    }
                 }
-                ColorOverlay {
-                    anchors.fill: imgFromWallet
-                    source: imgFromWallet
-                    color: root.fromAccount ? root.fromAccount.iconColor : Style.current.blue
+                SVGImage {
+                    id: fromArrow
+                    width: 13
+                    visible: typeof root.fromClicked === "function"
+                    anchors.right: parent.right
+                    anchors.rightMargin: 7
+                    anchors.verticalCenter: parent.verticalCenter
+                    fillMode: Image.PreserveAspectFit
+                    source: "../app/img/caret.svg"
+                    rotation: 270
+                    ColorOverlay {
+                        anchors.fill: parent
+                        visible: parent.visible
+                        source: parent
+                        color: Style.current.secondaryText
+                    }
+                }
```

Comment:
Can probably use `StatusIconButton` here ?

Reply:
Unfortunately not, the design is for a much smaller button, and also the MouseArea covers more than just the button.